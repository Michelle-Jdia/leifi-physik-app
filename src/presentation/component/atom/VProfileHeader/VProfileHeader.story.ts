import type { Meta, StoryFn } from '@storybook/vue3';
import VProfileHeader from '@/atom/VProfileHeader/VProfileHeader.vue';

export default {
    component: VProfileHeader,
} as Meta<typeof VProfileHeader>;

const Template: StoryFn<typeof VProfileHeader> = (args) => ({
    components: {
        VProfileHeader,
    },
    setup() {
        return {
            args,
        };
    },
    template: `
        <v-profileHeader v-bind="args">Profile header</v-profileHeader>
    `,
});

export const Base = Template.bind({});
