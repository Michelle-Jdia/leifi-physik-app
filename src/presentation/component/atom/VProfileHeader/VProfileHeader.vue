<script setup lang="ts">
import { IonImg } from '@ionic/vue';

interface Props {
    text?: string;
}

defineProps<Props>();
</script>

<template>
    <slot />

    <ion-img src="/image/component/profile-header.png" />
</template>
