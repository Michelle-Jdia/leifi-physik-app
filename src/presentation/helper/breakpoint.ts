export const breakpoints = {
    mobile: 576,
    tablet: 768,
};
