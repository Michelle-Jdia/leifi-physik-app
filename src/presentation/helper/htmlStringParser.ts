export const html =
    '<div class="rich-text-main"><div class="process_glossary"><p> </p><p>Zu vielen wichtigen Erkenntnissen über weißes Licht und seine Spektralfarben gelangte Newton mit einfachen Versuchen, die du <a href="/node/7678" target="_blank">hier</a>ausführlicher beschrieben findest.</p><h4 style="clear:both; margin-bottom:1em">Zerlegung von weißem Licht in die Spektralfarben</h4><figure role="group" class="rs_skip caption caption-drupal-entity"><div data-embed-button="media" data-entity-embed-display="view_mode:media.image_50_percent_center" data-entity-type="media" data-entity-uuid="73a22760-ff57-4a31-bac4-bc099cee2e72" class="align-right embedded-entity" data-langcode="de" data-entity-embed-display-settings="[]"><picture class=""><source srcset="https://www.app-api-dm4xeuy-bwyzh75a7hm5c.eu-2.platformsh.site/sites/default/files/images/fa62eb7a83830f00436cca015aa94f39/576spektralfarben_prismenspektrum_bild.svg" media="(max-width: 570px)" type="image/webp"><source srcset="https://www.app-api-dm4xeuy-bwyzh75a7hm5c.eu-2.platformsh.site/sites/default/files/images/fa62eb7a83830f00436cca015aa94f39/576spektralfarben_prismenspektrum_bild.svg" media="(max-width: 570px)" type="image/jpeg"><source srcset="https://www.app-api-dm4xeuy-bwyzh75a7hm5c.eu-2.platformsh.site/sites/default/files/images/fa62eb7a83830f00436cca015aa94f39/992spektralfarben_prismenspektrum_bild.svg" media="(min-width: 571px)" type="image/webp"><source srcset="https://www.app-api-dm4xeuy-bwyzh75a7hm5c.eu-2.platformsh.site/sites/default/files/images/fa62eb7a83830f00436cca015aa94f39/992spektralfarben_prismenspektrum_bild.svg" media="(min-width: 571px)" type="image/jpeg"><img src="https://www.app-api-dm4xeuy-bwyzh75a7hm5c.eu-2.platformsh.site/sites/default/files/images/fa62eb7a83830f00436cca015aa94f39/1000spektralfarben_prismenspektrum_bild.svg" alt="" /></source></source></source></source></picture><span class="image-copyright font-italic small rs_skip"><i class="far fa-copyright"></i><span class="copyright-text">Joachim Herz Stiftung</span></span></div><figcaption>Abb. 1 Zerlegung von weißem Licht in Spektralfarben mit Hilfe eines Prismas</figcaption></figure><p style="text-align:justify">Wenn du weißes Licht in geeigneter Weise auf ein Prisma fallen lässt, so kannst du auf einem Schirm dahinter beobachten, dass das weiße Licht in verschiedene Farben zerlegt wurde. Auf dem Schirm ist jeweils ein Bereich zu sehen, der rot, orange, gelb, grün, cyan, blau oder violett gefärbt ist. Diese Farben nennt man die <strong>Spektralfarben</strong>.</p><p style="text-align:justify">Die Spektralfarben kannst du auch bei einem Regenbogen finden. Hier ist die Sonne die Quelle für weißes Licht und die Regentropfen sorgen für die Zerlegung.</p><h4 style="clear:both; margin-bottom:1em">Unzerlegbarkeit der Spektralfarben</h4><figure role="group" class="rs_skip caption caption-drupal-entity"><div data-embed-button="media" data-entity-embed-display="view_mode:media.image_50_percent_center" data-entity-type="media" data-entity-uuid="69987120-eeaa-483c-9032-b04e0511d4c0" class="align-right embedded-entity" data-langcode="de" data-entity-embed-display-settings="[]"><picture class=""><source srcset="https://www.app-api-dm4xeuy-bwyzh75a7hm5c.eu-2.platformsh.site/sites/default/files/images/215ed628eaebf3845102ea5ade417e85/576spektralfarben_unzerlegbarkeit_bild.svg" media="(max-width: 570px)" type="image/webp"><source srcset="https://www.app-api-dm4xeuy-bwyzh75a7hm5c.eu-2.platformsh.site/sites/default/files/images/215ed628eaebf3845102ea5ade417e85/576spektralfarben_unzerlegbarkeit_bild.svg" media="(max-width: 570px)" type="image/jpeg"><source srcset="https://www.app-api-dm4xeuy-bwyzh75a7hm5c.eu-2.platformsh.site/sites/default/files/images/215ed628eaebf3845102ea5ade417e85/992spektralfarben_unzerlegbarkeit_bild.svg" media="(min-width: 571px)" type="image/webp"><source srcset="https://www.app-api-dm4xeuy-bwyzh75a7hm5c.eu-2.platformsh.site/sites/default/files/images/215ed628eaebf3845102ea5ade417e85/992spektralfarben_unzerlegbarkeit_bild.svg" media="(min-width: 571px)" type="image/jpeg"><img src="https://www.app-api-dm4xeuy-bwyzh75a7hm5c.eu-2.platformsh.site/sites/default/files/images/215ed628eaebf3845102ea5ade417e85/1000spektralfarben_unzerlegbarkeit_bild.svg" alt="" /></source></source></source></source></picture><span class="image-copyright font-italic small rs_skip"><i class="far fa-copyright"></i><span class="copyright-text">Joachim Herz Stiftung</span></span></div><figcaption>Abb. 2 Unzerlegbarkeit der Spektralfarben</figcaption></figure><p style="text-align:justify">Wenn du nun eine beliebige Spektralfarbe wieder durch ein Prisma fallen lässt, so siehst du auf dem Schirm, dass sich die Farbe des Lichtes nicht wieder geändert hat. Spektralfarben lassen sich nicht weiter zerlegen. Man nennt sie auch reine Farben.</p><h4 style="clear:both; margin-bottom:1em">Nicht sichtbare Strahlung im Spektrum</h4><p style="text-align:justify">Neben den Spektralfarben enthält zum Beispiel das Sonnenlicht noch weitere Anteile, die du nicht mit bloßem Auge sehen kannst - auch dann nicht, wenn du Sonnenlicht durch ein Prisma in seine Spektralfarben zerlegst.</p><p style="text-align:justify">Ein Anteil ist <strong>infrarote Strahlung</strong>(<strong>IR-Strahlung</strong>). Diese Strahlung wäre im Experiment auf dem Schirm (vgl. <strong>Abb. 1</strong>) oberhalb des roten Lichtes zu sehen. Infrarote Strahlung wird zum Beispiel bei Wärmelampen genutzt.</p><p style="text-align:justify">Ein weiterer nicht sichtbarer Teil wird als <strong>ultraviolette Strahlung</strong>(<strong>UV-Strahlung</strong>) bezeichnet. Diese Strahlung wäre im Experiment auf dem Schirm (vgl. <strong>Abb. 1</strong>) unterhalb des blauen Lichtes zu sehen. Ultraviolette Strahlung wird zum Beispiel zum Prüfen von Geldscheinen und zum Härten von Nagellack verwendet. Auch sind Teile der UV-Strahlung für den Sonnenbrand verantwortlich.</p></div></div><div class="clearfix"></div>';

const parser = new DOMParser();

export const doc = parser.parseFromString(html, 'text/html');
