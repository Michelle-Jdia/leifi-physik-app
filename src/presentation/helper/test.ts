import type { ComponentMountingOptions } from '@vue/test-utils';
import type { Component } from 'vue';
import { config, mount } from '@vue/test-utils';
import { expect, test } from 'vitest';

beforeAll(() => {
    config.global.renderStubDefaultSlot = true;
});

afterAll(() => {
    config.global.renderStubDefaultSlot = false;
});

export const testSnapshot = (
    component: Component,
    options?: ComponentMountingOptions<object>,
): void => {
    test('snapshot', () => {
        const wrapper = mount(component, {
            shallow: true,
            ...options,
        });

        expect(wrapper.element).toMatchSnapshot();
    });
};
