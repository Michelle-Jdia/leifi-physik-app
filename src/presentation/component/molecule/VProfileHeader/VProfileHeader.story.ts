import type { Meta, StoryFn } from '@storybook/vue3';
import VProfileHeader from '@/molecule/VProfileHeader/VProfileHeader.vue';

export default {
    component: VProfileHeader,
} as Meta<typeof VProfileHeader>;

const Template: StoryFn<typeof VProfileHeader> = (args) => ({
    components: {
        VProfileHeader,
    },
    setup() {
        return {
            args,
        };
    },
    template: `
        <v-profileHeader>
            Mia Mustermann
        </v-profileHeader>
    `,
});

export const Base = Template.bind({});
