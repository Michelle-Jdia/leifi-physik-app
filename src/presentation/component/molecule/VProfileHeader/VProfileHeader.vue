<script setup lang="ts">
import type { ClassValue } from 'clsx';
import { chevronForwardOutline } from 'ionicons/icons';
import { $t } from '@/application/lang/de';
import { cn } from '@/presentation/helper/style';
import VAvatar from '@/atom/avatar/VAvatar/VAvatar.vue';
import VIcon from '@/atom/icon/VIcon/VIcon.vue';

interface Props {
    modifier?: ClassValue;
}

defineProps<Props>();
</script>

<template>
    <div :class="cn('profile-header v-flex v-items-center', modifier)">
        <v-avatar data-testid="avatar-component" />

        <div class="v-ml-box-md v-mr-1">
            <h3 class="v-h3"><slot /></h3>

            <p class="v-h4 v-mt-1 v-font-fflight v-text-ty-helper">{{ $t.profile.showProfile }}</p>
        </div>

        <v-icon :icon="chevronForwardOutline" modifier="v-ml-auto" />
    </div>
</template>
