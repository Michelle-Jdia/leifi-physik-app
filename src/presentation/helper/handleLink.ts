import {
    isDownloadStringType,
    isIssueStringType,
    isTaskStringType,
} from '@/data/type/helper/contentType';
import { getContextualRouteNameByType } from '@/connection/helper/navigation';
import router from '@/connection/router';

function clickHandler(this: HTMLAnchorElement, event: MouseEvent) {
    const uuid = this.dataset.uuid;
    const type = this.dataset.type;

    if (
        !uuid ||
        !type ||
        !(isIssueStringType(type) || isDownloadStringType(type) || isTaskStringType(type))
    ) {
        return;
    }

    event.preventDefault();

    router.push({
        name: getContextualRouteNameByType(type),
        params: {
            id: uuid,
        },
    });

    this.removeEventListener('click', clickHandler);
}

export function handleLinksClick(containerElement: HTMLElement | null): void {
    if (!containerElement) {
        return;
    }

    const links = document.getElementsByTagName('a');

    Array.from(links).forEach((link: HTMLAnchorElement) => {
        link.addEventListener('click', clickHandler);
    });
}
