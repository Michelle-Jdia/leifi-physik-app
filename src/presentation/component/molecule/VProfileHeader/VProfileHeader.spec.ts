import { render } from '@testing-library/vue';
import { describe, test } from 'vitest';
import { testSnapshot } from '@/presentation/helper/test';
import VProfileHeader from '@/molecule/VProfileHeader/VProfileHeader.vue';

describe('VProfileHeader', () => {
    const options = {
        slot: {
            default: 'Profil anzeigen',
        },
        global: {
            stubs: {
                ionIcon: true,
            },
        },
    };

    test('renders correctly with default text, avatar, and slot', () => {
        const { getByText, getByTestId } = render(VProfileHeader, options);

        getByText('Profil anzeigen');

        const avatar = getByTestId('avatar-component');
        expect(avatar).toBeTruthy();

        getByText('Profil anzeigen');
    });

    test('renders correctly with custom slot text', () => {
        const { getByText } = render(VProfileHeader, {
            ...options,
            slots: {
                default: 'customSlotText',
            },
        });

        getByText('customSlotText');
    });

    testSnapshot(VProfileHeader, options);
});
