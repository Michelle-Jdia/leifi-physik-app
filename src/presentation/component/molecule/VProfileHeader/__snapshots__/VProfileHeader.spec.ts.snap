// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VProfileHeader > snapshot 1`] = `
<div
  class="profile-header v-flex v-items-center"
>
  <v-avatar-stub
    data-testid="avatar-component"
    modifier="false"
  />
  <div
    class="v-ml-box-md v-mr-1"
  >
    <h3
      class="v-h3"
    >
      
      
    </h3>
    <p
      class="v-h4 v-mt-1 v-font-fflight v-text-ty-helper"
    >
      Profil anzeigen
    </p>
  </div>
  <v-icon-stub
    icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><path stroke-linecap='round' stroke-linejoin='round' stroke-width='48' d='M184 112l144 144-144 144' class='ionicon-fill-none'/></svg>"
    modifier="v-ml-auto"
    size="default"
    withspaceleft="false"
    withspaceright="false"
  />
</div>
`;
